GeneralsMD/Code/GameEngine/Source/GameClient/InGameUI.cpp

```diff
@@ -1852,16 +1853,61 @@ void InGameUI::update( void )
 	Player* moneyPlayer = TheControlBar->getCurrentlyViewedPlayer();
 	if( moneyPlayer)
 	{
-		Int currentMoney = moneyPlayer->getMoney()->countMoney();
-		if( lastMoney != currentMoney )
+		Money *money = moneyPlayer->getMoney();
+		Bool showIncome = TheGlobalData->m_showMoneyPerMinute;
+		if (!showIncome)
 		{
-			UnicodeString buffer;
+			Int currentMoney = money->countMoney();
+			if( lastMoney != currentMoney )
+			{
+				UnicodeString buffer;
 
-			buffer.format( TheGameText->fetch( "GUI:ControlBarMoneyDisplay" ), currentMoney );
-			GadgetStaticTextSetText( moneyWin, buffer );
-			lastMoney = currentMoney;
+				buffer.format(TheGameText->fetch( "GUI:ControlBarMoneyDisplay" ), currentMoney );
+				GadgetStaticTextSetText( moneyWin, buffer );
+				lastMoney = currentMoney;
 
-		}  // end if
+			}  // end if
+		}
+		else
+		{
+			// TheSuperHackers @feature L3-M 21/08/2025 player money per minute
+			money->updateIncomeBucket();
+			UnsignedInt currentMoney = money->countMoney();
+			UnsignedInt cashPerMin = money->getCashPerMinute();
+			if (lastMoney != (Int)currentMoney || lastIncome != (Int)cashPerMin)
+			{
+				UnicodeString buffer;
+				UnicodeString moneyStr;
+				UnicodeString incomeStr;
+
+				if (currentMoney >= 100000)
```

Comment:
This code portion can become prettier by breaking out the money values formatting:

```cpp
UnicodeString moneyStr = formatMoneyValue(currentMoney);
UnicodeString incomeStr = formatIncomeValue(cashPerMin);
```